src/components/ImageItem/ImageItem.android.tsx

```diff
@@ -131,13 +133,8 @@ const ImageItem = ({
         onScrollEndDrag,
       })}
     >
-      <Animated.Image
-        {...panHandlers}
-        source={imageSrc}
-        style={imageStylesWithOpacity}
-        onLoad={onLoaded}
-      />
-      {(!isLoaded || !imageDimensions) && <ImageLoading />}
+      {CustomImageComponent ? (React.createElement(CustomImageComponent, {panHandlers:{...panHandlers}, source:{imageSrc}, style: {imageStylesWithOpacity}, onLoad:{onLoaded}, loadingImage: <ImageLoading />})) : (<Animated.Image {...panHandlers} source={imageSrc} style={imageStylesWithOpacity} onLoad={onLoaded}/>)}
```

Comment:
I have try your patch and I wonder if some typo exist when use `React.createElement`,
It might be `React.createElement(CustomImageComponent, {panHandlers:{...panHandlers}, source:imageSrc, style: imageStylesWithOpacity, onLoad:{onLoaded}, loadingImage: <ImageLoading />})`